### PR TITLE
Return 409 with error code TRANSIENT_ERROR when getting TransientExce…

### PR DESCRIPTION
**Pull request by hmusum**
…ption

Return transient error e.g. when creating load balancer fails. Return 409
with error code TRANSIENT_ERROR so that client know that request can be retried.

### CODE REVIEW

**Review by freva — COMMENTED** [review]
LB provisioner uses `LoadBalancerServiceException` in `node-repository`, maybe it should just be moved to `config-provisioning` and `HttpHandler` should catch that rather than the generic `TransientException` to follow the pattern set by the the certificate and dynamic provisioning?

The error code should also be added to https://github.com/vespa-engine/vespa/blob/f2aaaf89a0f24790ff2322b7b4f88256940c2917/controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/configserver/ConfigServerException.java#L29 and
https://github.com/vespa-engine/vespa/blob/f2aaaf89a0f24790ff2322b7b4f88256940c2917/controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/InternalStepRunner.java#L231
for it to actually be retried.

Also, we should consider moving those exceptions to separate package, no need for them to be public API?

**Comment by hmusum** [comment]
Thanks, good idea, please have a look now

File: configserver/src/test/java/com/yahoo/vespa/config/server/session/SessionPreparerTest.java [hunk 1]
```diff
@@ -327,4 +343,30 @@ private ApplicationId applicationId(String applicationName) {
                                   ApplicationName.from(applicationName), InstanceName.defaultName());
     }
 
+    private static class LoadBalancerException extends TransientException {
```
Comment (freva) [inline configserver/src/test/java/com/yahoo/vespa/config/server/session/SessionPreparerTest.java]: Use `LoadBalancerServiceException`

File: config-provisioning/src/main/java/com/yahoo/config/provision/exception/OutOfCapacityException.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
-// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
-package com.yahoo.config.provision;
+// Copyright 2019 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.config.provision.exception;
```
Comment (freva) [inline config-provisioning/src/main/java/com/yahoo/config/provision/exception/OutOfCapacityException.java]: Need to update the `provision-service` at the same time

File: configserver/src/test/java/com/yahoo/vespa/config/server/session/SessionPreparerTest.java (hunk 1 above)
Comment (hmusum) [reply configserver/src/test/java/com/yahoo/vespa/config/server/session/SessionPreparerTest.java]: Fixed

File: config-provisioning/src/main/java/com/yahoo/config/provision/exception/OutOfCapacityException.java (hunk 2 above)
Comment (hmusum) [reply config-provisioning/src/main/java/com/yahoo/config/provision/exception/OutOfCapacityException.java]: Fixed

**Review by freva — APPROVED** [review]
:100:

**Pull request merged**